### ISSUE TITLE
[REQUEST] Input only mode (no video)

### ISSUE THREAD

**Issue by samkitty** [ClassicOldSong/Apollo#177]
Would it be possible to add a setting to Apollo that disables video encoding/decoding and leaves the stream open for input (touch/mouse/controller) only? That way phones/handhelds can be used as a wireless gamepad, which would be very useful for something like party games with 8 players local co-op. Guests can simply show a pin code and start playing on their phones. Steam Link allows you to do this already by disabling video/audio and leaving just input. Apollo would replace the need for apps like [Unified Remote](https://play.google.com/store/apps/details?id=com.Relmtech.Remote) and [Remote Gamepad](https://play.google.com/store/apps/details?id=com.remotegamepad) if this feature is implemented.

This was a semi-popular [request](https://ideas.moonlight-stream.org/posts/106/forward-inputs-only-disable-video-audio-outputs) for the official Moonlight app but it was never worked on.

I tried launching a virtual display then turning it off in windows display settings, but the stream automatically switches to the primary display in that case. A truly black screen with input only (like when using External Monitor Mode) would be ideal.

Thanks!

**Comment by ClassicOldSong** [comment]
Possible but requires some heavy modifications. No guarantees to time of implementation.

**Comment by samkitty** [comment]
Got it. Current workaround is using tasker to display a black overlay on top of the stream, but that is a far from elegant solution.

Thank you for revitalizing sunshine/moonlight development 🫡

**Comment by ClassicOldSong** [comment]
Making an overlay doesn't stop the video/audio from being encoded and sent from the host. What's your purpose for not showing the copied view on other clients?

**Comment by samkitty** [comment]
> What's your purpose for not showing the copied view on other clients?

Using the phone as a trackpad/keyboard for the PC at night. The phone display becomes a distraction and might get OLED burn-in. I set the bitrate/resolution to lowest and mask it with Tasker. I know this isn't the same as not encoding the video in the first place, but it's the only way I can get a black screen on the client as far as I can tell.

**Comment by ClassicOldSong** [comment]
Hmmmm I see, I'll think about this.

**Comment by Wordsmith9091** [comment]
@samkitty This is in close proximity to yuor computer, right? Wouldn't it be easier to just use any of the several phone apps that can function as a trackpad and keyboard for a PC via bluetooth?

**Comment by samkitty** [comment]
> @samkitty This is in close proximity to yuor computer, right? Wouldn't it be easier to just use any of the several phone apps that can function as a trackpad and keyboard for a PC via bluetooth?

I've mentioned a few of said apps in my original post. Here are a few reasons why this makes sense: 

1- I use Moonlight to turn my android handheld into a wireless controller with gyro (no other app can do this) and the touch screen doubles as trackpad/full function keyboard which is perfect.

2- I often don't have enough controllers for party games and would like to allow guests to use their phones as wireless gamepads. There are apps that do this, but connecting different phones through a single Bluetooth dongle is unfeasible if not impossible. Moonlight can handle this with ease (up to 16 gamepads I believe).

3- All apps that offer mouse/keyboard for PC have a visible overlay which can't be turned off, this causes OLED burn in for extended use. Moonlight supports multitouch gestures, can run fullscreen and shows a fully black screen (with Simulate Secondary Display turned on in developer settings as a workaround for now, hopefully not for long!).

4- Bluetooth apps work only when in range so can't be used over the internet (i take my PC with me everywhere through streaming). And the only good wifi PC remote app **Unified Remote** has many limitations, like not being able to hold modifier keys (ctrl, alt, shift etc), can't disable autocorrect while typing or freezing when you hold backspace (key repeat rate bug) and more.

5- I already have Apollo running in the background at all times and would prefer to not have to keep other apps open like **Unified Remote** or **Remote Gamepad**.

As you can see, a robust and reliable solution like Moonlight is very much needed for this use case. I've tested every possible method of controlling a PC remotely and Moonlight is hands down the best way to do it, and it can be much better still (like connecting faster without video, staying connected in the background, faster polling rate for gyro, using less battery power while handling input only and so on). It's just so close to being perfect and I believe this feature can set Apollo/Artemis apart from the alternatives.

**Comment by lurebat** [comment]
Also useful in re: https://github.com/ClassicOldSong/moonlight-android/issues/97

Using the phone to connect a dualsense controller, but streaming on the tv.

**Comment by ClassicOldSong** [comment]
> Using the phone to connect a dualsense controller, but streaming on the tv.

Yeah a very valid point. Implenenting this is definetely on the list.

**Comment by ClassicOldSong** [comment]
Implemented in https://github.com/ClassicOldSong/Apollo/releases/tag/v0.2.9-alpha.4

**Comment by samkitty** [comment]
Awesome!

Initial impressions: this works extremely well. The server GPU usage is much lower and the client battery drain is reduced by ~2.7x, almost as good as Unified Remote! I haven't encountered any bugs either so this will be my main method of controlling my PC, even when I'm behind the desk it is nice to have a tablet as a touchpad while doing design work/drawing with stylus!

Thank you so much for putting the time and effort to get this done so quickly @ClassicOldSong ♡

![Image](https://github.com/user-attachments/assets/0c60eb25-b650-4130-9d6c-86ddceda6a08)
